control-plane-operator/controllers/hostedcontrolplane/cloud/azure/providerconfig.go

```diff
@@ -56,7 +73,7 @@ func azureConfigWithoutCredentials(hcp *hyperv1.HostedControlPlane, credentialsS
 		Location:                     hcp.Spec.Platform.Azure.Location,
 		VnetName:                     hcp.Spec.Platform.Azure.VnetName,
 		VnetResourceGroup:            hcp.Spec.Platform.Azure.ResourceGroupName,
-		SubnetName:                   hcp.Spec.Platform.Azure.SubnetName,
+		SubnetName:                   subnetName,
```

Comment:
The duality of name & ID seems superfluous - can we keep only the ID or only the name? Having to resort to string parsing to get one from the other seems fragile

Reply:
Agree but Azure cloud provider seems to only support subnetName - https://cloud-provider-azure.sigs.k8s.io/install/configs/

ARO HCP wants ID as much as possible in the API.

Reply:
If it's not possible to resolve that tension, are there other libraries we can use for parsing? `len(parts) != 11` feels fragile. I assume this kind of transformation is common for clients of Azure?

Reply:
Got a library from Ben Vessel. This has been updated.